src/main/java/me/zeroeightsix/kami/module/modules/render/WaypointRender.kt

```diff
@@ -62,40 +62,32 @@ object WaypointRender : Module() {
         INFO_BOX, ESP
     }
 
-    private val waypoints = ArrayList<WaypointInfo>()
+    private val waypointMap = TreeMap<BlockPos, TextComponent>(compareByDescending {
+        it.distanceSq(mc.player?.position ?: BlockPos(0, -69420, 0)) // This has to be sorted so the further ones doesn't overlaps the closer ones
```

Comment:
see above, just return if mc.player is null, there's no reason to render waypoints if you're not in game

Reply:
They are not rendering but they just be kept in the map. If for some reason the map gets modified when mc.player is null. It would cause a NPE when it tries to calculate distance for sorting

Reply:
yea, just return if mc.player is null

Reply:
That would be more codes than a single line of null check because there are multiple ways that the map get modified

Reply:
Alright, as long as it refreshes the list order once you login then it's fine